src/App/__tests__/App.test.tsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
```

Comment:
Habe hier mal noch ein paar beispielhafter Tests hinzugefügt. Mit der testing-library ist das relativ straightforward. Und zwar wird die Komponente einfach gerendert, und dann überprüft man, dass bestimmte Sachen gerendert wurden oder dass ein Klick auf eine bestimmte Komponente eben eine Veränderung im DOM auslöst. Das sind stark nutzergetriebene Tests, das heißt, es wird alles eher aus Sicht des Nutzers getestet, und nicht so sehr auf Code-Ebene (wie man es vor allem bei Unit Tests in Backend-Frameworks kennt). Darum kann man auch nicht direkt auf Komponenten zugreifen, sondern geht immer über den Text, der auch sichtbar ist.

Kurz als Beispiel:
Andere Testing-Libraries erlauben zum Beispiel, dass man einen Button in einer Komponente direkt auswählt, also `const link = render(App).get(GameLink)`. Da muss man aber halt wissen wie der Code der Komponente aussieht, und wenn sich was an der inneren Struktur ändert, (z.B. weil man GameLink in HeaderLink umbenennt), geht der Test kaputt.
Bei dieser Art des Testens geht man aber eben nicht auf die Komponenten selbst, sondern auf den Text, den der Nutzer auch sehen würde. Damit sind die Tests etwas abstrakter und nicht so fragil, und man kann eben auch wirklich das Nutzerverhalten nachspielen, und muss nicht auf Implementierungsdetails testen.